src/app/modules/form/admin-form/admin-form.service.ts

```diff
@@ -1148,3 +1164,172 @@ const isMobileFieldUpdateAllowed = (
+  })
+
+  return ResultAsync.fromPromise(
+    mongoose.connection.transaction(async (session: ClientSession) => {
+      await FormModel.updateMsgSrvcName(form._id, msgSrvcName, session)
+      await secretsManager.createSecret(body).promise()
+    }),
+    (error) => {
+      logger.error({
+        message: 'Error encountered when creating Twilio Secret',
+        meta: {
+          action: 'createTwilio',
+          body,
+        },
+        error,
+      })
+
+      return new ApplicationError('Error occurred when creating Twilio!')
+    },
+  )
+}
+
+export const updateTwilioCredentials = (
+  msgSrvcName: string,
+  twilioCredentials: TwilioCredentials,
+): ResultAsync<PutSecretValueResponse, DatabaseError> => {
+  if (!isCredentialsValid(twilioCredentials))
+    return errAsync(new MalformedParametersError('Credentials are invalid!'))
+
+  const twilioCredentialsData: TwilioCredentialsData =
+    new TwilioCredentialsData(twilioCredentials)
+
+  const body: SecretsManager.Types.PutSecretValueRequest = {
+    SecretId: msgSrvcName,
+    SecretString: twilioCredentialsData.toString(),
+  }
+
+  return ResultAsync.fromPromise(
+    secretsManager.getSecretValue({ SecretId: msgSrvcName }).promise(),
+    (error) => {
+      logger.error({
+        message: 'Twilio Credentials do not exist in Secrets Manager',
+        meta: {
+          action: 'updateTwilioCredentials',
+          msgSrvcName,
+        },
+        error,
+      })
+
+      return new DatabaseError(
+        'Twilio Credentials do not exist in Secrets Manager',
+      )
+    },
+  ).andThen(() => {
+    logger.info({
+      message: 'Twilio Credentials has been found in Secrets Manager',
+      meta: {
+        action: 'updateTwilioCredentials',
+        msgSrvcName,
+      },
+    })
+
+    twilioClientCache.del(msgSrvcName)
+
+    return ResultAsync.fromPromise(
+      secretsManager.putSecretValue(body).promise(),
+      (error) => {
+        logger.error({
+          message: 'Error occurred when updating Twilio in Secret Manager!',
+          meta: {
+            action: 'updateTwilioCredentials',
+            body,
+          },
+          error,
+        })
+
+        return new DatabaseError(
+          'Error occurred when updating Twilio in Secret Manager!',
+        )
+      },
+    )
+  })
+}
+
+export const deleteTwilioCredentials = (
+  formId: string,
+  msgSrvcName: string,
+): ResultAsync<unknown, ApplicationError> => {
+  const body: SecretsManager.Types.DeleteSecretRequest = {
+    SecretId: msgSrvcName,
+    ForceDeleteWithoutRecovery: true,
+  }
+
+  return ResultAsync.fromPromise(
+    secretsManager
+      .getSecretValue({
+        SecretId: msgSrvcName,
+      })
+      .promise(),
+    (error) => {
+      logger.error({
+        message: 'Twilio Credentials do not exist in Secrets Manager',
+        meta: {
+          action: 'deleteTwilioCredentials',
+          msgSrvcName,
+        },
+        error,
+      })
+
+      return new DatabaseError(
```

Comment:
I think we don't need to throw an error here, we can just return a success immediately if the credentials already do not exist? This follows the thinking that deletion technically "succeeds" when the object to delete is deleted in the first place

Reply:
You're right, fixed it!